### PR TITLE
Add PHP 8.1 support

### PR DESCRIPTION
**Pull request by Miljoen**
Remove deprecated Serializable interface, add mixed return type to jsonSerialize()

### CODE REVIEW

**Comment by Miljoen** [comment]
@czim https://www.php.net/manual/en/language.types.declarations.php

The `mixed` return type is a breaking change.
The same goes for the `Serializable` interface removal.

@czim Can you make a branch for a new major release to merge this PR to instead?

File: composer.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "keywords": [ "file", "attachment", "laravel", "upload" ],
     "homepage": "https://github.com/czim",
     "require": {
-        "php": "^7.4|^8.0",
+        "php": ">=8.0.0",
```
Comment (czim) [inline composer.json:7]: I would prefer `"^8.0"`, bit cleaner notation, does the same and does not assume it'll work in PHP 9 ;]

File: src/Attachment/Attachment.php [hunk 2]
```diff
@@ -1139,14 +1138,10 @@ protected function getEventDispatcher()
         return app('events');
     }
 
-
-    /**
-     * @return string
-     */
-    public function serialize()
+    public function __serialize(): array
```
Comment (czim) [inline src/Attachment/Attachment.php:1141]: Would appreciate an iterable type docblock: `@return array<string, mixed>`

File: src/Attachment/Attachment.php [hunk 3]
```diff
@@ -479,9 +478,9 @@ public function variantContentType($variant)
     /**
      * Return a JSON representation of this class.
      *
-     * @return array
+     * @return mixed
```
Comment (czim) [inline src/Attachment/Attachment.php:481]: This docblock may be removed

File: composer.json (hunk 1 above)
Comment (Miljoen) [reply composer.json]: Makes sense, changed.

File: src/Attachment/Attachment.php (hunk 3 above)
Comment (Miljoen) [reply src/Attachment/Attachment.php]: Docblock removed

File: src/Attachment/Attachment.php (hunk 2 above)
Comment (Miljoen) [reply src/Attachment/Attachment.php:1139]: Docblock has been added

**Pull request merged**